### PR TITLE
Refer explicitly to Elastic License v2

### PR DESCRIPTION
**Pull request by jsanz**
As discussed in the Kibana issue https://github.com/elastic/kibana/issues/102909 the current license statement in our `package.json` is needing an unnecessary override https://github.com/elastic/kibana/blob/master/src/dev/license_checker/config.ts#L76

This change should also make the license explicit in npm.js

image

### CODE REVIEW

**Review by thomasneirynck — APPROVED** [review]
thx!

**Comment by jsanz** [comment]
One thing @thomasneirynck, when running `yarn lint` we get a warning message about the License

```text
$ yarn lint
yarn run v1.22.10
warning package.json: License should be a valid SPDX license expression
$ tsc --noEmit
$ eslint . --ext .ts,.tsx
Done in 5.83s.
```

Not sure if this would prevent the package to be published but it's definitely something to consider and maybe the reason for the current value for the license. @nickpeihl will probably know more about this and I would be more comfortable waiting for him to review this before merging and publishing a new patch version of the client. What do you think?

I guess this is something we will eventually see also in EUI as well?

**Comment by thomasneirynck** [comment]
@jsanz yes, let's wait to merge until we understand this lint-error. thx!

**Comment by jsanz** [comment]
For awareness, EUI has adopted the `SEE LICENSE IN LICENSE.txt` text in the library `package.json` file with the adoption of the Elastic License v2

https://github.com/elastic/eui/pull/4930/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

**Comment by jsanz** [comment]
I'm closing this since we are adopting the same approach as EUI. We can reopen if we find a better way to handle the Elastic License for npm packages.

**Pull request closed without merging**
